tests/cc/test_usdt_args.cc fails to compile on armv7hl

`parser` is only defined for `aarch64`, `ppc64`, `s390x` and `x86_64`, so when compiling for `armv7hl` it fails with
```
/builddir/build/BUILD/bcc/tests/cc/test_usdt_args.cc:67:14: error: 'parser' was not declared in this scope; did you mean 'pause'?
   67 |     REQUIRE(!parser.parse(&arg));
      |              ^~~~~~
/builddir/build/BUILD/bcc/tests/cc/test_usdt_args.cc:67:14: error: 'parser' was not declared in this scope; did you mean 'pause'?
   67 |     REQUIRE(!parser.parse(&arg));
      |              ^~~~~~
/builddir/build/BUILD/bcc/tests/cc/test_usdt_args.cc:205:13: error: 'parser' was not declared in this scope; did you mean 'pause'?
  205 |     REQUIRE(parser.done());
      |             ^~~~~~
/builddir/build/BUILD/bcc/tests/cc/test_usdt_args.cc:205:13: error: 'parser' was not declared in this scope; did you mean 'pause'?
  205 |     REQUIRE(parser.done());
      |             ^~~~~~
```

--- a/tests/cc/test_usdt_args.cc
+++ b/tests/cc/test_usdt_args.cc
@@ -52,6 +52,10 @@ static void verify_register(USDT::ArgumentParser &parser, int arg_size,
   REQUIRE(arg.scale() == scale);
 }
 
+/* supported arches only */
+#if defined(__aarch64__) || defined(__powerpc64__) || \
+    defined(__s390x__) || defined(__x86_64__)
+
 TEST_CASE("test usdt argument parsing", "[usdt]") {
   SECTION("parse failure") {
 #ifdef __aarch64__
@@ -205,3 +209,5 @@ TEST_CASE("test usdt argument parsing", "[usdt]") {
     REQUIRE(parser.done());
   }
 }
+
+#endif /* supported arches only */

